pkg/commands/app.go

```diff
@@ -1214,6 +1216,13 @@ func NewSBOMCommand(globalFlags *flag.GlobalFlagGroup) *cobra.Command {
 				return xerrors.Errorf("flag error: %w", err)
 			}
 
+			// For SBOM-to-SBOM scanning (for example, to add vulnerabilities to the SBOM file), we should not modify the scanned file.
+			// cf. https://github.com/aquasecurity/trivy/pull/9439#issuecomment-3295533665
+			if slices.Contains(types.SupportedSBOMFormats, options.Format) &&
+				(!slices.Equal(options.PkgTypes, types.PkgTypes) || !slices.Equal(options.PkgRelationships, ftypes.Relationships)) {
+				log.Warnf("Trivy doesn't support '--pkg-types' and '--pkg-relationships' options for SBOM to SBOM scan. These options will be ignored.")
+			}
+
```

Comment:
I added warning about the `--pkg-types` and `--pkg-relationship` flags for SBOM-to-SBOM scanning (as we discussed [here](https://github.com/aquasecurity/trivy/pull/9439#issuecomment-3295533665)).

I am not sure this is the good place, but for other places (e.g. in [Align](https://github.com/aquasecurity/trivy/blob/6e99dd304c7fad8213489039e7ca42909383b5ff/pkg/flag/options.go#L416) function) we would need to throw `targetKind`.
However, as we said, this is most likely a very rare case, so I settled on this solution.

Reply:
Will fix the linter issues